cmd/zc_enumerator.go

```diff
@@ -443,7 +447,7 @@ func InitResourceTraverser(resource common.ResourceString, location common.Locat
 		} else if listOfVersionIds != nil {
 			output = newBlobVersionsTraverser(resourceURL, *p, *ctx, recursive, includeDirectoryStubs, incrementEnumerationCounter, listOfVersionIds, cpkOptions)
 		} else {
-			output = newBlobTraverser(resourceURL, *p, *ctx, recursive, includeDirectoryStubs, incrementEnumerationCounter, s2sPreserveBlobTags, cpkOptions, includeDeleted, includeSnapshot, includeVersion)
+			output = newBlobTraverser(resourceURL, *p, *ctx, recursive, includeDirectoryStubs, incrementEnumerationCounter, s2sPreserveBlobTags, cpkOptions, includeDeleted, includeSnapshot, includeVersion, getProperties)
```

Comment:
This method appears to be growing in terms of flags, shall we think to put option bag here instead of passing down so many parameters?

Reply:
Yes I was thinking the same, not sure if this is a design decision on Azcopy's end or if there hasn't been time to refactor this. I can do it if the team is on board for it.

Reply:
I'm on board with using an options bag as well.

Reply:
Option bag would be more clean. We've had the list growing forever.

Reply:
I like the options bag idea too!